src/vivarium/interface/utilities.py

```diff
@@ -105,9 +107,43 @@ def wrapped_method(*args, **kwargs):
     return method_wrapper
 
 
-def get_output_root(results_directory, model_specification_file):
+def get_output_location_like_string(
+    artifact_path: Union[str, Path],
+    model_spec_path: Union[str, Path],
+) -> str:
+    """Find a good string to use as model_name or location in output path creation.
+
+    Parameters
+    ----------
+    artifact_path
+        Path to the artifact file, if exists, else should be None
+    model_spec_path
+        Path to the model specification file. This must exist.
+
+    Returns
+    -------
+    str
+        A location-like string for use in output labeling.
+    """
+    if artifact_path:
+        location = Path(artifact_path).stem
```

Comment:
There's no reason beyond convention that an artifact is going to be labelled by a location.  Locations are a (GBD) public health specific identifier and have no real place in `vivarium` proper.  What you're creating here is a formal logic for model names which says first use the artifact name and if that doesn't exist, use the model spec name.  None of that is specific to locations.

Reply:
Is the concern with the word "location"? Would the concern be satisfied if I did `s/location/location_like_thing/g`?

Reply:
Yes.  "Location" has is not a formal concept in vivarium.  It's a formal concept in GBD.  `vivarium` is a generic simulation framework and we don't want leaky boundaries.

Reply:
You want something like `model_name`